app/models/user.rb

```diff
@@ -144,6 +146,20 @@ def permissible_zip
     end
   end
 
+  def permissible_state
+    if self.has_role? :admin
+      true # admins might live anywhere
+    elsif self.state.blank?
+      true # no state is allowed
+    else
+      if User::VALID_STATES.map{|s| s[0].to_s}.include?( self.state.to_s.upcase )
```

Comment:
You don't need the map, you could just use VALID_STATES.keys.include? since your keys are already strings.